### PR TITLE
Feature casttimefix

### PR DESCRIPTION
**Pull request by dagochen**
redone the fix from last PR, with little improvements

since everybody ignored the first PR, this might also be something for @Warlockbugs

### CODE REVIEW

**Comment by Warlockbugs** [comment]
I am afraid, we will have to start from the beginning on the issue then.
A correctly described problem usually contains the solution within description. 😄 

Original PR link: https://github.com/cmangos/mangos-classic/pull/187

> Aimed Shot casting time is decreased by 0,5 seconds for no reason.

There is no such thing as no reason. 😄 Debugging is needed. DBC entry is set to 3s, so this is core shenanigans.

Now lets move on to ammo issue.

You mentioned that Aimed Shot and Multi-shoot should not receive cast time bonus from ammo. But what about other shots, including those which came with newer expansions (Steady, etc)?

File: src/game/SpellMgr.cpp [hunk 1]
```diff
@@ -120,23 +122,36 @@ uint32 GetSpellCastTime(SpellEntry const* spellInfo, Spell const* spell)
 
     int32 castTime = spellCastTimeEntry->CastTime;
 
+    // Multishot only (alternative set CastingTimeIndex = 3)
+    if (spellInfo->HasAttribute(SPELL_ATTR_RANGED) && (!spell || !spell->IsAutoRepeat()) && spellInfo->SpellFamilyFlags & 0x0000000000001000) 
+        castTime = 500;
```
Comment (dagochen) [inline src/game/SpellMgr.cpp]: yes

**Comment by dagochen** [comment]
Oh did I wrote decreased? Ofcourse its an increase of 0.5 seconds...someone wanted to increase multishot and did that for EVERY ranged spell. 

And about the steady shot thing I am not sure, because its TBC and then all this mechanics work completely different, considering delay of autoshots and haste on items and so on....

**Comment by Warlockbugs** [comment]
I need it to know if all shots (i.e. ranged weapon abilities which use ammo), do not benefit from quiver's speed. In this case there is a fairly straightforward way to handle it.

**Comment by dagochen** [comment]
You propably need to check for the spell effects on haste-gear in tbc/wotlk and other spells like Heroism, which improved casting time for steady shot. If there is still a difference between "Apply Aura: Haste - Ranged (Ammo Only)" (quivers in classic) and "Apply Aura: Haste - Ranged" (non-quiver-effects), then this fix should work fine.

**Comment by dagochen** [comment]
And later on quivers have been removed, which might help too?

**Comment by cyberium** [comment]
Only part of this is merged. The rest need more research/work/proof.

**Pull request closed without merging**